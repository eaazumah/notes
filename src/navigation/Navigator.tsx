import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import React from 'react';
import {routes as r} from '../@shared/constants';
import {navigatorRef, onNavigatorReady} from '../@shared/services/navigator';
import Editor from '../editor/Editor';
import Splash from '../onboarding/Splash';
import Settings from '../settings/Settings';
import BottomTab from './BottomTab';

const routes = [
  {
    name: r.splash,
    component: Splash,
  },
  {
    name: r.editor,
    component: Editor,
  },
  {
    name: r.bottomTab,
    component: BottomTab,
  },
  {
    name: r.settings,
    component: Settings,
  },
];

const Stack = createStackNavigator();

const Navigator: React.FC = () => {
  return (
    <NavigationContainer ref={navigatorRef} onReady={onNavigatorReady}>
      <Stack.Navigator screenOptions={{headerShown: false}}>
        {routes.map((route, index) => (
          <Stack.Screen
            key={index}
            name={route.name}
            component={route.component}
          />
        ))}
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default Navigator;
