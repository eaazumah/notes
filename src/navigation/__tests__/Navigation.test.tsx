jest.useFakeTimers();
jest.setTimeout(100000);

import {cleanup, render} from '@testing-library/react-native';
import React from 'react';
import TestProvider from '../../__testUtils__/Providers';
import Navigator from '../Navigator';

describe('Navigator', () => {
  afterEach(cleanup);
  test('Navigator renders correctly', () => {
    const navigator = render(
      <TestProvider>
        <Navigator />
      </TestProvider>,
    );

    expect(navigator.toJSON()).toMatchSnapshot();
  });

  test('Navigator functionality test', async () => {
    const navigator = render(
      <TestProvider>
        <Navigator />
      </TestProvider>,
    );

    // const {queryByText, queryByTestId} = navigator;
    // const headerTitle = queryByText('Notes');
    // expect(headerTitle).toBeTruthy();

    // const openSetting = queryByTestId('open-setting');
    // expect(openSetting).toBeTruthy();

    // fireEvent.press(openSetting!);
    // const themeComponent = queryByText('theme');
    // expect(themeComponent).toBeTruthy();
  });
});
