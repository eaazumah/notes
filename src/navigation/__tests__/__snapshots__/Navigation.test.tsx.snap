// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator Navigator renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              nativeID="animatedComponent"
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#F5F5F5",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <SvgMock
                          height={200}
                          width={200}
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
